js/dialog.js

```diff
@@ -113,12 +94,11 @@
   });
 
   // close error dialog window ------------------------------------------------
-
   document.addEventListener('click', function (evt) {
     var target = evt.target;
-    var div = document.querySelector('.modal-error');
+    var modal = document.querySelector('.modal-error');
     if (target.className === 'modal-error' || target.className === 'modal-error__close') {
```

Comment:
если у модального окна добавить модификатор, например 'modal-error--small', то окно уже не закроется, поэтому лучше проверять наличие класса, а не что класс есть и он только один
```suggestion
    if (target.classList.contains('modal-error') || target.classList.contains('modal-error__close') {
```